src/components/visualizations/chart/highcharts/plugins/autohideLabels/test/autohideColumnLabels.spec.ts

```diff
@@ -32,3 +33,42 @@ describe("getStackLabelPointsForDualAxis", () => {
         ]);
     });
 });
+
+describe("isOverlappingWidth", () => {
+    const visiblePointsWithoutShape: IDataPoint[] = [
+        {
+            dataLabel: {
+                width: 98,
+                padding: 2,
+            },
+        },
+    ];
+
+    const visiblePointsWithShape: IDataPoint[] = [
+        {
+            ...visiblePointsWithoutShape[0],
+            shapeArgs: {
+                width: 105,
+            },
+        },
+    ];
+
+    it("should return true when point has datalabel width greater than shape width", () => {
+        const visiblePointsWithShape: IDataPoint[] = [
+            {
+                ...visiblePointsWithoutShape[0],
+                shapeArgs: {
+                    width: 100,
+                },
+            },
+        ];
+        expect(isOverlappingWidth(visiblePointsWithShape)).toEqual(true);
+    });
+
+    it.each([[false, visiblePointsWithoutShape], [false, visiblePointsWithShape]])(
```

Comment:
Since both cases return false, you could simplify this to something like

```ts
it.each([visiblePointsWithoutShape, visiblePointsWithShape])(
    "should return false when point has datalabel width less than shape width or shapeArgs is undefined",
    // ...
```

Reply:
Also, I would argue making two separate tests would be clearer.

```ts
it("should return false when point has datalabel width less than shape width", () => {
    // case with visiblePointsWithShape
});

it("should return false when shapeArgs is undefined", () => {
    // case with visiblePointsWithoutShape
});
```

Reply:
Updated